gbpservice/nfp/tests/unit/nfp/configurator/test_firewall.py

```diff
@@ -0,0 +1,887 @@
+    def __init__(self):
+        fo = FakeObjects()
+        kwargs = fo._fake_kwargs()
+        self.data = {
+                    'context': {'notification_data': {},
+                                'resource': 'firewall'},
+                    'firewall': fo._fake_firewall_obj(),
+                    'host': fo.host,
+                    'kwargs': kwargs,
+                    'vm_mgmt_ip': fo.vm_mgmt_ip,
+                    'service_vendor': fo.service_vendor,
+                    'source_cidrs': fo.source_cidrs,
+                    'destination_cidr': fo.destination_cidr,
+                    'gateway_ip': fo.gateway_ip,
+                    'provider_interface_position': (
+                                        fo.provider_interface_position)}
+        self.id = 'dummy'
+
+
+class GenericConfigEventHandlerTestCase(unittest.TestCase):
+    ''' Generic Config Handler for Firewall module '''
+
+    def __init__(self, *args, **kwargs):
+        super(GenericConfigEventHandlerTestCase, self).__init__(
+                                                        *args, **kwargs)
+        self.fo = FakeObjects()
+        self.empty = self.fo.empty_dict
+        self.context = {'notification_data': {},
+                        'resource': 'interfaces'}
+
+    @mock.patch(__name__ + '.FakeObjects.nqueue')
+    @mock.patch(__name__ + '.FakeObjects.rpcmgr')
+    @mock.patch(__name__ + '.FakeObjects.drivers')
+    @mock.patch(__name__ + '.FakeObjects.sc')
+    def _get_GenericConfigEventHandler_object(self, sc,
+                                              drivers, rpcmgr, nqueue):
+        agent = gc.GenericConfigEventHandler(sc, drivers, rpcmgr, nqueue)
+        return agent, sc
+
+    def _test_handle_event(self, ev):
+        agent, sc = self._get_GenericConfigEventHandler_object()
+        driver = FwaasDriver()
+        with mock.patch.object(
+                driver, 'configure_interfaces') as mock_config_inte, \
+             mock.patch.object(
+                driver, 'clear_interfaces') as mock_clear_inte, \
+             mock.patch.object(
+                driver, 'configure_routes') as mock_config_src_routes, \
+             mock.patch.object(
+                driver, 'clear_routes') as mock_delete_src_routes, \
+             mock.patch.object(
+                sc, 'poll_event') as mock_hm_poll_event, \
+             mock.patch.object(
+                sc, 'poll_event_done') as mock_hm_poll_event_done, \
+             mock.patch.object(
+                driver, 'configure_healthmonitor', return_value='SUCCESS') as mock_hm_driver_call, \
+             mock.patch.object(
+                agent, '_get_driver', return_value=driver):
+            agent.handle_event(ev)
+            
+            kwargs = self.fo._fake_kwargs()
+            kwargs.pop('request_info')
+            if ev.id == 'CONFIGURE_INTERFACES':
+                mock_config_inte.assert_called_with(
+                                        self.empty, kwargs)
```

Comment:
Is the context always empty when we invoke the driver API? Can we double check this?